services/PackageReader.ts

```diff
@@ -0,0 +1,163 @@
+import * as fs from "fs/promises";
+import * as path from "path";
+
+/**
+ * Reads information from a file which adhere to control file format of Debian Policy.
+ * The provided control file should be organized in paragraph where each paragraph
+ * represents a package information. Paragraphs are separated from each other from a empty new line.
+ *
+ * Within a paragraph information is organized in (field, value) pairs. Lines within the paragraph start
+ * with the name of the field a column, space and then the values.
+ * Simple fields have their value within the same line, while folded and multiline fields, have their value span multiple lines
+ * The addional lines of a multiline field start with a whitespace.
+ *
+ * The class deals with the relationship fields in a different way by giving the posibility to remove versions
+ * from the list of pakckages contained in the value of relationship fields.
+ *
+ *
+ */
+
+const paragraphDelimiter = "\n\n";
+const fieldLineDelimiter = "\n";
+const packageField = "Package";
+const versionField = "Version";
+const descriptionField = "Description";
+const dependsField = "Depends";
+
+const packageStatusFile =
+  process.env.PACKAGE_STATUS_FILE || path.resolve(".", "packages.status");
+
+const packageFileContent = fs.readFile(packageStatusFile).then((doc) => {
+  console.log("Reading");
+  return doc.toString();
+});
+
+const searchableField: (fieldName: string) => string = (fieldName) =>
+  fieldName + ": ";
+
+const readSimpleFieldFromParagraph: (
+  packageParagraph: string,
+  fieldName: string
+) => string = (packageParagraph, fieldName) => {
+  const _fieldName = searchableField(fieldName);
+
+  const fieldValue = packageParagraph
+    .split(fieldLineDelimiter)
+    .filter((line) => line.startsWith(_fieldName))
+    .map((line) => line.replace(_fieldName, ""));
+  if (fieldValue) {
+    return fieldValue[0];
+  } else {
+    return "";
+  }
+};
```

Comment:
You can use a destructor here:

```suggestion
  const [fieldValue] = packageParagraph
    .split(fieldLineDelimiter)
    .filter((line) => line.startsWith(_fieldName))
    .map((line) => line.replace(_fieldName, ""));
  
  return fieldValue ?? "";
};
```

Reply:
Yes this is cleaner and nicer agree. A question though. There is just a single variable there: `fieldValue` 
Memory wise in situations when memory is critical, is there any implication if instead of just returning that variable you construct an array holding that single one just for the purpose on using the nullish coalescing operator?

This is more out of curiosity because I like how the code looks with having to deconstruct it this way.

Reply:
I don't think there will be a memory performance difference. And if there is, it's negligible. First off, the code that you write is not necessarily 1:1 the code that you deploy, thanks to Webpack and its possible optimizing preprocessors. Secondly, the JavaScript engines are extremely good at optimizing code during runtime (very similar to JVM's JIT), so these kinds of optimizations are not really worth even keeping in mind.

Anyways, I don't exactly know how the internals work, but I would expect that this code will always create multiple arrays, no matter how it's written: first the result of `split`, then `filter` and finally `map`. The resulting array from `map` will be garbage collected in a similar pace anyways, since the function exists very quickly afterwards.

Always go for the most readable version of your code first. Only when it's proven to be a problem, start doing optimizations.